File: src/main/java/com/oshi/libgenericsettings/data/TitleSwitchData.java
package com.oshi.libgenericsettings.data;

import android.support.annotation.ColorRes;

/**
 * Title & Switch. Being use in layout/view_type_title_switch.xml
 */
public class TitleSwitchData extends TitleData {

    protected boolean isSwitchOn;
    protected @ColorRes int switchTrackCheckedColor = 0;
    protected @ColorRes int switchTrackUncheckedColor = 0;
    protected @ColorRes int switchThumbCheckedColor = 0;
    protected @ColorRes int switchThumbUncheckedColor = 0;

    public TitleSwitchData(String title) {
        super(title);
    }

    public TitleSwitchData(String title, boolean isSwitchOn) {
        super(title);
        this.title = title;
        this.isSwitchOn = isSwitchOn;
    }

    @Override
    public int getViewType() {
        return VIEW_TYPE_SETTINGS_TITLE_SWITCH;
    }

    public boolean isSwitchOn() {
        return isSwitchOn;
    }

    public void setSwitchOn(boolean switchOn) {
        isSwitchOn = switchOn;
    }

    public int getSwitchTrackCheckedColor() {
        return switchTrackCheckedColor;
    }

    public int getSwitchTrackUncheckedColor() {
        return switchTrackUncheckedColor;
    }

    public int getSwitchThumbCheckedColor() {
        return switchThumbCheckedColor;
    }

    public int getSwitchThumbUncheckedColor() {
        return switchThumbUncheckedColor;
    }

    public void setSwitchTrackCheckedColor(int switchTrackCheckedColor) {
        this.switchTrackCheckedColor = switchTrackCheckedColor;
    }

    public void setSwitchTrackUncheckedColor(int switchTrackUncheckedColor) {
        this.switchTrackUncheckedColor = switchTrackUncheckedColor;
    }

    public void setSwitchThumbCheckedColor(int switchThumbCheckedColor) {
        this.switchThumbCheckedColor = switchThumbCheckedColor;
    }

    public void setSwitchThumbUncheckedColor(int switchThumbUncheckedColor) {
        this.switchThumbUncheckedColor = switchThumbUncheckedColor;
    }
}

File: src/main/java/com/oshi/libgenericsettings/data/TitleSwitchData.kt
package com.oshi.libgenericsettings.data

import android.support.annotation.ColorRes
import com.oshi.libgenericsettings.data.BaseViewTypeData.Companion.SettingsViewTypes

/**
 * Title & Switch. Being use in layout/view_type_title_switch.xml
 */
open class TitleSwitchData(title : String) : TitleData(title) {

    var isSwitchOn : Boolean = false
    @ColorRes var switchTrackCheckedColor = 0
    @ColorRes var switchTrackUncheckedColor = 0
    @ColorRes var switchThumbCheckedColor = 0
    @ColorRes var switchThumbUncheckedColor = 0

    override fun getViewType(): SettingsViewTypes {
        return SettingsViewTypes.VIEW_TYPE_SETTINGS_TITLE_SWITCH
    }

}